source/common/upstream/cluster_manager_impl.cc

```diff
@@ -923,13 +923,10 @@ ClusterManagerImpl::loadCluster(const envoy::config::cluster::v3::Cluster& clust
 
   // Check if the cluster provided load balancing policy is used. We need handle it as special
   // case.
-  bool cluster_provided_lb = cluster_info->lbType() == LoadBalancerType::ClusterProvided;
-  if (cluster_info->lbType() == LoadBalancerType::LoadBalancingPolicyConfig) {
-    TypedLoadBalancerFactory* typed_lb_factory = cluster_info->loadBalancerFactory();
-    RELEASE_ASSERT(typed_lb_factory != nullptr, "ClusterInfo should contain a valid factory");
-    cluster_provided_lb =
-        typed_lb_factory->name() == "envoy.load_balancing_policies.cluster_provided";
-  }
+  TypedLoadBalancerFactory* typed_lb_factory = cluster_info->loadBalancerFactory();
+  RELEASE_ASSERT(typed_lb_factory != nullptr, "ClusterInfo should contain a valid factory");
```

Comment:
we're really working to discourage release asserts.  Can you comment how this extension is hard-linked into Envoy binaries such that this is a reasonable guarantee?